app/src/main/java/com/brandon/forzenbook/viewmodels/LoginViewModel.kt

```diff
@@ -0,0 +1,96 @@
+package com.brandon.forzenbook.viewmodels
+
+import android.net.ConnectivityManager
+import android.util.Log
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.mutableStateOf
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.brandon.forzenbook.usecase.GetLoginCodeUseCase
+import com.brandon.forzenbook.usecase.LoginUseCase
+import com.brandon.forzenbook.usecase.LoginValidationState
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+@HiltViewModel
+class LoginViewModel @Inject constructor(
+    private val loginUseCase: LoginUseCase,
+    private val getLoginCodeUseCase: GetLoginCodeUseCase,
+    private val connectivityManager: ConnectivityManager,
+) : ViewModel() {
+
+    private val _uiState: MutableState<LoginUiState> = mutableStateOf(LoginUiState.Idle())
+    val uiState: MutableState<LoginUiState>
+        get() = _uiState
+
+    sealed interface LoginUiState {
+
+        data class Idle(
+            val email: String = "",
+            val isEmailError: Boolean = false,
+            val isGenericError: Boolean = false,
+            val isCodeSent: Boolean = false,
+        ) : LoginUiState
+
+        data class Loading(
+            val email: String = "",
+            val isCodeSent: Boolean = false,
+        ) : LoginUiState
+
+        // TODO remove this once we have another screen FA-83
+        object LoginWithCode : LoginUiState
+
+    }
+
+    fun checkInternetConnection(): Boolean {
+        return connectivityManager.activeNetworkInfo?.isConnected ?: false
```

Comment:
nit: You could make a connectivity manager extension function to do this logic so you don't need to duplicate it

Reply:
good callout! created extension function for this